src/pages/base.js

```diff
@@ -14,13 +15,20 @@ class Page {
   }
 
   visitWithParameters(data) {
-    let url = this.url;
-
-    for (const item of data.raw()) {
-      url = url.replace(`:${item[0]}`, item[1]);
-    }
+    const additionalParams = [];
+
+    const url =
+      data.raw().reduce((prev, item) => {
```

Comment:
I think it would be a good approach to change the arguments names to e.g. `(url, param)`
as the `prev` is not really the "prev" value and the naming is kinda confusing

Reply:
url is already in use